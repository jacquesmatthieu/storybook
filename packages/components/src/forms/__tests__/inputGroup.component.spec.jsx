import React from 'react';
import { shallow } from 'enzyme';

import InputGroup from '../inputGroup.component';

describe('InputGroup', () => {
  let props;

  beforeEach(() => {
    props = {
      name: 'foo',
    };
  });

  const getWrapper = () => shallow(<InputGroup {...props}>Bar</InputGroup>);

  it('should render with default props', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with specified props', () => {
    // Given
    props.label = 'bar';

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
