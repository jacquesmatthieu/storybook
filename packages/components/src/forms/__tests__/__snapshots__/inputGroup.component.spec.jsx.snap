// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render with default props 1`] = `
<inputGroupcomponent__InputGroupWrapper
  className=""
  fullWidth={false}
  variant="input"
>
  <typographycomponent__Typography
    htmlFor="foo"
    variant="inputLabel"
  />
  Bar
</inputGroupcomponent__InputGroupWrapper>
`;

exports[`InputGroup should render with specified props 1`] = `
<inputGroupcomponent__InputGroupWrapper
  className=""
  fullWidth={false}
  variant="input"
>
  <typographycomponent__Typography
    htmlFor="foo"
    variant="inputLabel"
  >
    bar
  </typographycomponent__Typography>
  Bar
</inputGroupcomponent__InputGroupWrapper>
`;
